lua/fzf-lua-frecency/init.lua

```diff
@@ -1,145 +1,193 @@
-  local algo = require "fzf-lua-frecency.algo"
-
-  local wrapped_enter = function(action)
-    return function(selected, action_opts)
-      vim.schedule(function()
-        for _, sel in ipairs(selected) do
-          if display_score then
-            sel = h.strip_score(sel)
-          end
-          -- https://github.com/ibhagwan/fzf-lua/wiki/Advanced#explore-changes-from-a-git-branch
-          local filename = fzf_lua.path.entry_to_file(sel, action_opts).path
-          algo.update_file_score(filename, {
-            update_type = "increase",
-            cwd = cwd,
-            db_dir = db_dir,
-            fd_cmd = fd_cmd,
-          })
+  if vim.fn.executable "fdfind" == 1 then
+    cmd = string.format("fdfind %s", fd_args)
+  elseif vim.fn.executable "fd" == 1 then
+    cmd = string.format("fd %s", fd_args)
+  elseif vim.fn.executable "rg" == 1 then
+    -- return [[rg --files -g "\!.git" "$(pwd)"]]
+    cmd = string.format([[rg --files -g "!.git"]], opts.cmd or vim.uv.cwd())
+  else
+    FzfLua.utils.warn "[fzf-lua-frecency] 'all_files' requires 'fd' or 'rg'."
+    return nil
+  end
+  for k, v in pairs {
+    follow = opts.toggle_follow_flag or "-L",
+    hidden = opts.toggle_hidden_flag or "--hidden",
+    no_ignore = opts.toggle_ignore_flag or "--no-ignore",
+  } do
+    (function()
+      local toggle, is_find = opts[k], nil
+      -- Do nothing unless opt was set
+      if opts[k] == nil then return end
+      if cmd:match "^dir" then return end
+      if cmd:match "^find" then
+        if k == "no_ignore" then return end
+        if k == "hidden" then
+          is_find = true
+          toggle = not opts[k]
+          v = [[\! -path '*/.*']]
         end
-      end)
-
-      return action(selected, action_opts)
-    end
+      end
+      cmd = FzfLua.utils.toggle_cmd_flag(cmd, v, toggle, is_find)
+    end)()
   end
+  return cmd
+end
 
-  local actions = vim.tbl_deep_extend("force", fzf_lua.defaults.actions.files, {
-    enter = wrapped_enter(fzf_lua.defaults.actions.files.enter),
-    ["ctrl-x"] = {
-      fn = function(selected, action_opts)
-        for _, sel in ipairs(selected) do
-          if display_score then
-            sel = h.strip_score(sel)
-          end
-
-          local filename = fzf_lua.path.entry_to_file(sel, action_opts).path
-          algo.update_file_score(filename, {
-            update_type = "remove",
-            cwd = cwd,
-            db_dir = db_dir,
-            fd_cmd = fd_cmd,
-          })
-        end
-      end,
-      reload = true,
-    },
+--- @diagnostic disable-next-line: unused-local
+M.setup = function(opts)
+  -- Singleton setup
+  if M._did_setup then return end
+  M._did_setup = true
+
+  -- Trigger lazy loading if need be, creates the FzfLua global object
+  require "fzf-lua"
+
+  -- Register as an fzf-lua extenstion, merge default opts with users' setup opts
+  FzfLua.register_extension("frecency", M.frecency, vim.tbl_deep_extend("keep", opts or {}, {
```

Comment:
Do you think `force` would be better than `keep` to prefer the user's opts vs the default?

Reply:
> Do you think force would be better than keep to prefer the user's opts vs the default? (This assumes the opts from setup are passed along to frecency)

Setup options would lose their relevance in such case, imagine `setup({ display_score=false })` hiding the score globally, setting this to `force` would override the users' wishes and restore the defaults.

Reply:
Ah my bad I misread this code, looks good!